core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/project/ProjectResource.scala

```diff
@@ -348,6 +348,27 @@ class ProjectResource {
     }
   }
 
+  /**
+    * This method updates the description of a specified, existing project
+    *
+    * @param pid project ID
+    */
+  @POST
+  @Path("/{pid}/update/description")
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def updateProjectDescription(@PathParam("pid") pid: UInteger, project: UserProject): Unit = {
```

Comment:
please add `SessionUser` and manages permission to this endpoint.

Reply:
Done, added logic to verify project exists / user has access to it for all relevant endpoints.